[14.0][MIG] sql_export

florian-dacosta opened the pull request:
Standard migration

florian-dacosta committed:
Add sql export module (migration from v7)

Add rollback after executing query as a double security with blacklist terms

add known issue in readme

bguillot committed:
[IMP] add support of variables in queries

[FIX] encoding with mogrify

[FIX] hide placeholder

mourad-ehm committed:
[MIG] Migrate module from 8.0 to 9.0

legalsylvain committed:
[REF] create a new module sql_request_abstract

florian-dacosta committed:
Fix version

AaronHForgeFlow committed:
[IMP]encode output file

legalsylvain committed:
[FIX] various improvement. Courtesy @dreispt review

oca-transbot committed:
OCA Transbot updated translations from Transifex

AaronHForgeFlow committed:
[MIG][10.0]sql_export v10

MiquelRForgeFlow committed:
[FIX] mogrify doesn't allow dicts

oca-travis committed:
[UPD] Update sql_export.pot

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (50 of 50 strings)

Translation: server-tools-10.0/server-tools-10.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-10-0/server-tools-10-0-sql_export/de/

bealdav committed:
IMP Readme for Sql Export Module

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

bealdav committed:
PORT v12 sql_export

bealdav committed:
FIX sql_export: test and misc

oca-travis committed:
[UPD] Update sql_export.pot

sbidoul committed:
[FIX] title level in readme fragments

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/

bealdav committed:
IMP sql_export: render field as require when matching field def

florian-dacosta committed:
Fix uses of params in sql query

OCA-git-bot committed:
[UPD] README.rst

florian-dacosta committed:
Adapt module sql_export_mail to be compatible with multiple file format

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (49 of 49 strings)

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/zh_CN/

florian-dacosta committed:
Migrate sql_export_excel to v12

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (49 of 49 strings)

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/pt_BR/

florian-dacosta committed:
Small fixes for v12/python3 standard

Co-Authored-By: David Beal <david.beal@akretion.com>

oca-travis committed:
[UPD] Update sql_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/

oca-travis committed:
[UPD] Update sql_export.pot

OCA-git-bot committed:
sql_export 12.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/

fernandahf committed:
[FIX] sql_export: fix warning two fields have same label

oca-travis committed:
[UPD] Update sql_export.pot

OCA-git-bot committed:
sql_export 12.0.1.1.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-sql_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-sql_export/

florian-dacosta committed:
[IMP] sql_export: black, isort, prettier

florian-dacosta committed:
Migrate to 14.0

petrus-v commented on sql_export/README.rst:
```diff
@@ -0,0 +1,120 @@
+  since variables features has been introduced. This can be fixed by
+  overloading _prepare_request_check_execution() function.
+
+* Move modules sql_request_abstract and sql_export to oca/reporting-engine
```
I'm wondering when is the best time to attempt this kind of works ?

I don't know if discussion already take place somewhere ?

 `sql_request_abstract` is already merged in branch 14.0 so I don't know what's the best to do. At least tell when it's the best time to do it and makes some consistent with `sql_request_abstract` README  (also not sure what's the OCA best practice for that kind of cross ref do it in different PR or in the same atomic commit ?)

petrus-v reviewed (changes requested):
Quick review and test few query on run bot (I'havn't plays with access write) LGTM once future is clarify

Other few remarks that I won't block that could be done after migration:
- [ ] it would be nice display a message in the execute query wizard if there are not parameters or directly execute and download the file
- [ ] I don't really understand states on sql.export model as long you can still edit the sql query after validation maybe move that field to readonly ?
- [ ] it would be nice to add in the query field help how to use parameters

florian-dacosta commented:
Hello @petrus-v 
The improvement proposal you made seem good to me. 
As you said it could be done after or during the migration, I have no time right now but will keep it in mind of if someone want to contribute this.
About the repo, I still believe it should move to report engine, as it really is a report, but I also think it should stays with sql_request_abstract. (so maybe do it in a next version? Not sure though...

petrus-v commented:
> Hello @petrus-v
> The improvement proposal you made seem good to me.
> As you said it could be done after or during the migration, I have no time right now but will keep it in mind of if someone want to contribute this.
> About the repo, I still believe it should move to report engine, as it really is a report, but I also think it should stays with sql_request_abstract. (so maybe do it in a next version? Not sure though...

At least (I believe) a choice should be token and explicit in the readme to avoid loosing time at the next migration.

I suppose one of the @OCA/tools-maintainers team have the authority to make the decision !

> I don't really understand states on sql.export model as long you can still edit the sql query after validation maybe move that field to readonly

And I think it would be nice to clarify that point to be consistency, probably move the sql query to readonly once it has been validated.

florian-dacosta committed:
Update roadmap for v15 and make some usability improvements
Improve usability about variable creation and choice by hiding base wizard base fields and display simplified view
Add a description on paramter fields to help the user to use parameters
If not parameter are used for the query, display the wizard with the generated file directly

florian-dacosta commented:
@petrus-v 
I've added a Roadmap on `sql_request_abstract` so that the repo change is not forgotten on next version.

>And I think it would be nice to clarify that point to be consistency, probably move the sql query to readonly once it has been validated.
You are right, I've change the view and put some fields in readonly mode once validated.

I've also made some usability change about parameters and added a description on the field.
There are still a lot of room for improvement  thought.

petrus-v commented on sql_export/models/sql_export.py:
```diff
@@ -27,7 +27,22 @@ class SqlExport(models.Model):
-        domain=[("model", "=", "sql.file.wizard")],
+        domain=[("model", "=", "sql.file.wizard"), ("state", "=", "manual")],
+        help="Before adding parameters, make sure you have created one that fill your "
+        "need in the dedicated menu with the right type dans label. \n"
```
in label ?

florian-dacosta replied:
Miss typing "and"
Thanks!

petrus-v commented on sql_export/wizard/wizard_file.py:
```diff
@@ -70,12 +70,10 @@ def export_sql(self):
-            company_id = self.env.context.get(
-                "force_company", self.env.user.company_id.id
-            )
+            company_id = self.env.company.id
```
I suppose that the force_user/ force_company was to be able to create dedicated views according action menu context ie: one dedicated view per company (instead of the current company) can be confused at some point, not sure what's the best todo !

florian-dacosta replied:
No, these are kind of special parameters, you can use in the sql query %(company_id)s so it will be replaced by the company of the user.
Same for the user_id

It may be usefull to have a dynamic query, that depends on the user and/or of the company of the user, that execute it.
So, the same query won't give the same csv file depending on which user execute it.

The force_company context is deprecated now in favor of the current company on the environment

But indeed, the sql export menu does not manage multi company for now, we may add a company_id field on it and add some ir.rules, in order to display only the queries made for the allowed companies of the user, it would be a nice improvement for intercompany projects

petrus-v reviewed (approved):
A note and type otherwise you get my :+1: 

thanks for migration

bealdav reviewed (approved):
LGTM

dreispt commented:
/ocabot merge nobump
